src/miniscript/iter.rs

```diff
@@ -0,0 +1,577 @@
+                vec![k[0]],
+                vec![h[1]],
+                false,
+            ),
+            (
+                ms_str!("and_b(c:pk_k({}),sjtv:sha256({}))", k[0], sha256_hash),
+                vec![k[0]],
+                vec![],
+                false,
+            ),
+            (
+                ms_str!(
+                    "andor(c:pk_k({}),jtv:sha256({}),c:pk_h({}))",
+                    k[1],
+                    sha256_hash,
+                    h[2]
+                ),
+                vec![k[1]],
+                vec![h[2]],
+                false,
+            ),
+            (
+                ms_str!("multi(3,{},{},{},{},{})", k[9], k[8], k[7], k[0], k[1]),
+                vec![k[9], k[8], k[7], k[0], k[1]],
+                vec![],
+                true,
+            ),
+            (
+                ms_str!(
+                    "thresh(3,c:pk_k({}),sc:pk_k({}),sc:pk_k({}),sc:pk_k({}),sc:pk_k({}))",
+                    k[2],
+                    k[3],
+                    k[4],
+                    k[5],
+                    k[6]
+                ),
+                vec![k[2], k[3], k[4], k[5], k[6]],
+                vec![],
+                false,
+            ),
+            (
+                ms_str!(
+                    "or_d(multi(2,{},{}),and_v(v:multi(2,{},{}),older(10000)))",
+                    k[6],
+                    k[7],
+                    k[8],
+                    k[9]
+                ),
+                vec![k[6], k[7], k[8], k[9]],
+                vec![],
+                false,
+            ),
+            (
+                ms_str!(
+                    "or_d(multi(3,{},{},{},{},{}),\
+                      and_v(v:thresh(2,c:pk_h({}),\
+                      ac:pk_h({}),ac:pk_h({})),older(10000)))",
+                    k[0],
+                    k[2],
+                    k[4],
+                    k[6],
+                    k[9],
+                    h[8],
+                    h[7],
+                    h[0]
+                ),
+                vec![k[0], k[2], k[4], k[6], k[9]],
+                vec![h[8], h[7], h[0]],
+                false,
+            ),
+        ]
+    }
+
+    #[test]
+    fn get_keys() {
+        gen_testcases()
+            .into_iter()
+            .enumerate()
+            .for_each(|(idx, (ms, k, _, test_top_level))| {
+                if !test_top_level {
+                    return;
+                }
+                println!("Test case {}: {:?}", idx + 1, ms);
+                let ms = *ms.branches().first().unwrap_or(&&ms);
+                assert_eq!(ms.get_leaf_pubkeys(), k);
+            })
+    }
+
+    #[test]
+    fn get_hashes() {
+        gen_testcases()
+            .into_iter()
+            .enumerate()
+            .for_each(|(idx, (ms, k, h, test_top_level))| {
+                if !test_top_level {
+                    return;
+                }
+                println!("Test case {}: {:?}", idx + 1, ms);
```

Comment:
Can we remove the `println` from tests? Currently, no other tests do any printing, it would be good to stick to that.

Reply:
Sorry, missed that. Will do